libs/wingc/src/fold.rs

```diff
@@ -413,26 +413,71 @@ pub fn fold_type_annotation<F>(f: &mut F, node: TypeAnnotation) -> TypeAnnotatio
 where
 	F: Fold + ?Sized,
 {
-	match node {
-		TypeAnnotation::Number => TypeAnnotation::Number,
-		TypeAnnotation::String => TypeAnnotation::String,
-		TypeAnnotation::Bool => TypeAnnotation::Bool,
-		TypeAnnotation::Duration => TypeAnnotation::Duration,
-		TypeAnnotation::Json => TypeAnnotation::Json,
-		TypeAnnotation::MutJson => TypeAnnotation::MutJson,
-		TypeAnnotation::Optional(t) => TypeAnnotation::Optional(Box::new(f.fold_type_annotation(*t))),
-		TypeAnnotation::Array(t) => TypeAnnotation::Array(Box::new(f.fold_type_annotation(*t))),
-		TypeAnnotation::MutArray(t) => TypeAnnotation::MutArray(Box::new(f.fold_type_annotation(*t))),
-		TypeAnnotation::Map(t) => TypeAnnotation::Map(Box::new(f.fold_type_annotation(*t))),
-		TypeAnnotation::MutMap(t) => TypeAnnotation::MutMap(Box::new(f.fold_type_annotation(*t))),
-		TypeAnnotation::Set(t) => TypeAnnotation::Set(Box::new(f.fold_type_annotation(*t))),
-		TypeAnnotation::MutSet(t) => TypeAnnotation::MutSet(Box::new(f.fold_type_annotation(*t))),
-		TypeAnnotation::Function(t) => TypeAnnotation::Function(FunctionTypeAnnotation {
-			param_types: t.param_types.into_iter().map(|t| f.fold_type_annotation(t)).collect(),
-			return_type: t.return_type.map(|t| Box::new(f.fold_type_annotation(*t))),
-			phase: t.phase,
-		}),
-		TypeAnnotation::UserDefined(t) => TypeAnnotation::UserDefined(f.fold_user_defined_type(t)),
+	match node.kind {
+		TypeAnnotationKind::Number => TypeAnnotation {
+			kind: TypeAnnotationKind::Number,
+			span: node.span,
+		},
```

Comment:
Small simplification to avoid having to copy `span` all of those times
```rust
	let kind = match node.kind {
		TypeAnnotationKind::Number => TypeAnnotationKind::Number,
		TypeAnnotationKind::String => TypeAnnotationKind::String,
		...
	};
	TypeAnnotation {
	 	kind,
	 	span: node.span,
	 }
```